addons/base_vat/models/res_partner.py

```diff
@@ -222,7 +222,14 @@ def _build_vat_error_message(self, country_code, wrong_vat, record_label):
         else:
             company = self.env.company
 
-        vat_label = company.country_id.vat_label if company.country_id else "Tax ID"
+        country = self.env['res.country'].search([('code', '=ilike', country_code)], limit=1)
```

Comment:
```suggestion
        country = self.env['res.country'].search([('code', '=', country_code.upper())], limit=1)
```

same like https://github.com/odoo/odoo/blob/9db18b55b541831be8a21a2276b35a9befd32558/addons/base_vat/models/res_partner.py#L198